frontend/src/app/components/MainInstallButton.js

```diff
@@ -49,11 +56,8 @@ export default class MainInstallButton extends React.Component {
       </div>
     );
   }
-
-  install() {
-    const { eventCategory, hasAddon } = this.props;
-    if (hasAddon) { return; }
-    this.setState({ isInstalling: true });
-    installAddon(eventCategory);
-  }
 }
+
+MainInstallButton.contextTypes = {
```

Comment:
Actually, is this what you meant with this bit in `<App>`?
```javascript
{React.cloneElement(this.props.children, ...this.props)}
```
Could we supply a generic `installAddon` helper function in a prop from `<App>` and leave this component ignorant of the Redux store? Happy not to block this PR and save that as a future improvement if so.